pkg/entrypoint/entrypointer.go

```diff
@@ -184,31 +184,36 @@ func (e Entrypointer) Go() error {
 	// strings.Split(..) with an empty string returns an array that contains one element, an empty string.
 	// This creates an error when trying to open the result folder as a file.
 	if len(e.Results) >= 1 && e.Results[0] != "" {
-		if err := e.readResultsFromDisk(); err != nil {
+		if err := e.readResultsFromDisk(pipeline.DefaultResultPath); err != nil {
 			logger.Fatalf("Error while handling results: %s", err)
 		}
 	}
 
 	return err
 }
 
-func (e Entrypointer) readResultsFromDisk() error {
+func (e Entrypointer) readResultsFromDisk(resultDir string) error {
 	output := []v1beta1.PipelineResourceResult{}
 	for _, resultFile := range e.Results {
 		if resultFile == "" {
 			continue
 		}
-		fileContents, err := ioutil.ReadFile(filepath.Join(pipeline.DefaultResultPath, resultFile))
+		fileContents, err := ioutil.ReadFile(filepath.Join(resultDir, resultFile))
 		if os.IsNotExist(err) {
 			continue
 		} else if err != nil {
 			return err
 		}
+		// unmarshal the filesContents to string or array
+		aos := v1beta1.ArrayOrString{}
+		if err := aos.UnmarshalJSON(fileContents); err != nil {
+			return err
+		}
 		// if the file doesn't exist, ignore it
 		output = append(output, v1beta1.PipelineResourceResult{
-			Key:        resultFile,
-			Value:      string(fileContents),
-			ResultType: v1beta1.TaskRunResultType,
+			Key:         resultFile,
+			ResultValue: &aos,
+			ResultType:  v1beta1.TaskRunResultType,
```

Comment:
Seems like to me you can do:
```
tmp := &v1beta1.PipelineResourceResult{
			Key:         resultFile,
			ResultValue: v1beta1.ArrayOrString{},
			ResultType:  v1beta1.TaskRunResultType,
}
...
if err := *tmp.ResultValue.UnmarshalJSON(fileContents); err != nil {
			return err
		}
...
output = append(output, *tmp)
```
This makes it possible to set ResultValue to be ArrayOrString instead of *ArrayOrString

Reply:
if that we, we need to have. There's no way to avoid  *ArrayOrString. Another reason we need to use pointer for ResultValue is that we need to use omitempty for json marshalling, and only pointer can work
```
tmp := &v1beta1.PipelineResourceResult{
			Key:         resultFile,
			ResultValue: &v1beta1.ArrayOrString{},
			ResultType:  v1beta1.TaskRunResultType,
}
```